octoprint_firmwareupdate/__init__.py

```diff
@@ -101,6 +107,33 @@ def on_after_startup(self):
         else:
             self._logger.info("Auto firmware update disabled, skipping...")
 
+    @octoprint.plugin.BlueprintPlugin.route("/upload", methods=["POST"])
+    @restricted_access
+    @admin_permission.require(403)
+    def upload_file(self):
+        # Delete any firmware files that may exist when using custom firmware
+        self._delete_firmware_files()
+        if "base64String" not in flask.request.values:
+            return flask.make_response("Expected a base64String value", 400)
+
+        try:
+            decode = base64.b64decode(flask.request.values['base64String'])
+            self._check_directories()
+            with open(os.path.join(self.firmware_directory,
```

Comment:
probably should `try`/`except` this, in-case the file doesn't exist, or the write fails, or other numerous issues